docs: Added note about GBS and jsonl samples to DPO tutorial (#345)
NVIDIA/NeMo-Aligner: docs/user-guide/dpo.rst

Always follow the prompt-response template format used during your SFT training for DPO, as failure to do so will produce a model which outputs garbage text. You should create one jsonl file in the format above for your training data and one jsonl for your validation data.

> Please note: your jsonl file needs to have at least as many samples as the Global Batch Size you plan to use during training. So if your GBS=64, ensure that both your training and validation files have at least 64 samples. Using a file which has fewer samples than GBS will result in a crash.

Reviewer: Applying some rules the tech doc team had asked for:

```suggestion
Your JSONL file must contain at least as many samples as the Global Batch Size (GBS) you plan to use during training. For example, if GBS = 64, ensure that both your training and validation files include at least 64 samples. Using a file with fewer samples than the GBS will result in a crash.
```
